node/src/accountant/mod.rs

```diff
@@ -2078,208 +2576,303 @@ mod tests {
+                .lock()
+                .unwrap();
+        assert_eq!(
+            *scan_for_new_payables_notify_later_params,
+            vec![(
+                ScanForNewPayables {
+                    response_skeleton_opt: None
+                },
+                new_payable_expected_computed_interval
+            )]
+        );
+        let mut compute_interval_params = compute_interval_params_arc.lock().unwrap();
+        let (p_scheduling_now, last_new_payable_scan_timestamp, _) =
+            compute_interval_params.remove(0);
+        assert_eq!(last_new_payable_scan_timestamp, UNIX_EPOCH);
+        let scan_for_new_payables_notify_params =
+            scan_for_new_payables_notify_params_arc.lock().unwrap();
+        assert!(
+            scan_for_new_payables_notify_params.is_empty(),
+            "We did not expect any immediate scheduling of new payables, but it happened {:?}",
+            scan_for_new_payables_notify_params
+        );
+        let scan_for_retry_payables_notify_params =
+            scan_for_retry_payables_notify_params_arc.lock().unwrap();
+        assert!(
+            scan_for_retry_payables_notify_params.is_empty(),
+            "We did not expect any scheduling of retry payables, but it happened {:?}",
+            scan_for_retry_payables_notify_params
+        );
+        // Assertions on the receivable scanner proper functioning
+        let mut receivable_start_scan_params = receivable_start_scan_params_arc.lock().unwrap();
+        let (r_wallet, _r_started_at, r_response_skeleton_opt, r_logger, r_trigger_msg_name_str) =
+            receivable_start_scan_params.remove(0);
+        assert_eq!(r_wallet, earning_wallet);
+        assert_eq!(r_response_skeleton_opt, None);
+        assert!(
+            r_trigger_msg_name_str.contains("Receivable"),
+            "Should contain Receivable but {}",
+            r_trigger_msg_name_str
+        );
+        assert!(
+            receivable_start_scan_params.is_empty(),
+            "Should be already empty but was {:?}",
+            receivable_start_scan_params
+        );
+        test_use_of_the_same_logger(&r_logger, test_name, Some("r"));
+        let scan_for_receivables_notify_later_params =
+            scan_for_receivables_notify_later_params_arc.lock().unwrap();
+        assert_eq!(
+            *scan_for_receivables_notify_later_params,
+            vec![(
+                ScanForReceivables {
+                    response_skeleton_opt: None
+                },
+                receivable_scan_interval
+            )]
+        );
+        // Finally, an assertion to prove course of actions in time. We expect the pending payable
+        // scanner to take place before the payable scanner.
+        // I do believe it's impossible that these two instants would happen at the same time, until
+        // this is run on a real super-giga-computer.
+        assert!(pp_scan_started_at < p_scheduling_now);
     }
```

Comment:
You'd like to write a new test which would take care of the pending transactions found in the initial scan unlike the test which has no such issue. Since, both of these tests are meant for the situation when the accountant starts up, it would be a good idea to combine most of it's functionality through a helper function (in case it seems a possibility).